### PR TITLE
feat: ➕ add StrictExtract

### LINKED ISSUE #211: StrictExtract for getting Union elements
I posted the idea here – https://twitter.com/beraliv/status/1425767234878812165

In short, I want to implement `StrictExtract`:

```ts
type StrictExtract<T, U extends Partial<T>> = T extends U ? T : never;
```

And then it will help identifying TypeScript errors if I misspelled or mistyped the key or the value

Maybe we need to think about `undefined` here but don't see any other problems

WDYT?

### PR DESCRIPTION
**Pull request by Beraliv**
Fixes #211

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -330,6 +332,56 @@ type SimplifiedComplexObjectWithOmit = Omit<ComplexObject, "nested" | "simple" |
...
+}
+
+interface Cat {
+  type: "cat";
+  meow(): void;
+}
+
+interface Mouse {
+  type: "mouse";
+  squaek(): void;
```
Comment (ritschwumm) [inline README.md:352]: typo, should be squeak

File: test/index.ts [hunk 2]
```diff
@@ -655,3 +656,32 @@ function testIsTuple() {
...
+  }
+
+  interface Cat {
+    type: "cat";
+    meow(): void;
+  }
+
+  interface Mouse {
+    type: "mouse";
+    squaek(): void;
```
Comment (ritschwumm) [inline test/index.ts:673]: squeak

**Review by krzkaczor — APPROVED** [review]
Missing changeset

**Comment by Beraliv** [comment]
> Missing changeset

I will add it before the merge

File: README.md (hunk 1 above)
Comment (Beraliv) [reply README.md:352]: Thank you!

Changed ✅

File: test/index.ts (hunk 2 above)
Comment (Beraliv) [reply test/index.ts:673]: Thank you, changed! ✅

**Pull request merged**